lib/Meta/Metas.php

```diff
@@ -125,4 +144,72 @@ private function findByTitle(string $text): ?MetaEntry
 
         return null;
     }
+
+    /** @return array<string, LinkTarget> */
+    public function getLinkTargets(): array
+    {
+        return $this->linkTargets;
+    }
+
+    public function setLinkTarget(string $name, LinkTarget $linkTarget): void
+    {
+        if (array_key_exists($name, $this->linkTargets)) {
+            $this->errorManager->warning('Duplicate anchor ".. _' . $linkTarget->getName() . '" found.');
+            $i = 2;
+            while (array_key_exists($name . '-' . $i, $this->linkTargets)) {
+                $i++;
+            }
+
+            $name .= '-' . $i;
+            $linkTarget->setName($name);
+            $linkTarget->setDuplicate(true);
```

Comment:
If I dump `$this->linkTargets` right before this `if` closes, I have different results:

In the case where it passes, the title is "Headline one", otherwise it's "Headline two", so it looks like you're right about this being an issue with the processing order.

Reply:
All the documents are put into a queue for processing. Maybe on some machines these run on parallel processes or something like that? Anyhow we do not promise a certain order and renaming a link target that caused a warning here.

Reply:
And indeed, when I dump `$file` inside `ParseQueueProcessor::process()`, I get either `index` or `duplicateAnchor`.

Reply:
So should this issue concern us? Is it a bug or a feature?

Reply:
But as it turns out, this might not be an order issue because when I dump `$this->fileInfos` after the foreach loop in `Scanner::scan()`, I get 1 file vs 2: `duplicateAnchor` is not found when the test passes.

Reply:
in the output folder however duplicateAnchor.html gets generated and I explicitely load it in the test

Reply:
I do have both files in the Scanner:scan() $this->fileInfos after the for loop and I have passing tests...

Reply:
Ah my work computer was not up to date with the latest changes. Now it works…

Reply:
Oh but it works on the CI too, I didn't even notice that :facepalm: